Solver: Cache Concretization Results (#48198)
LLNL/spack: lib/spack/docs/config_yaml.rst

   aliases:
     inst: install -v

-------------------------------
``concretization_cache:enable``
-------------------------------

> When set to ``true``, Spack will create a cache for the solver output of

Reviewer: ```suggestion
When set to ``true``, Spack will utilize a cache of solver outputs from
```

successful concretization runs. When enabled, Spack will check the concretization

> cache prior to running the solver, and if a previous request to solve a given

Reviewer: ```suggestion
cache prior to running the solver. If a previous request to solve a given
```

solver data from the cache rather than running the solver. Specs not previously
concretized will be added to the cache on a successful solve. The cache additionally
holds solver statistics, so commands like ``spack solve`` will still return information
about the run that produced a given solver result.

This cache is a subcache of the :ref:`Misc Cache` and as such will be cleaned when the Misc
Cache is cleaned.

When ``false`` or ommitted, all concretization requests will be performed from scatch

----------------------------
``concretization_cache:url``
----------------------------

Path to the location where Spack will root the concretization cache. Currently this only supports
paths on the local filesystem.

Default location is under the :ref:`Misc Cache` at: ``$misc_cache/concretization``

------------------------------------
``concretization_cache:entry_limit``
------------------------------------

Sets a limit on the number of concretization results that Spack will cache. The limit is evaluated
after each concretization run; if Spack has stored more results than the limit allows, the
oldest concretization results are pruned until 10% of the limit has been removed.

> Setting this value to 0 or -1 disables the automatic pruning. It is expected users will be

Reviewer: Let's pick just one value. I think I prefer `0`, and we can set the range in the schema to be `"minimum": 0,`.

responsible for maintaining this cache.

-----------------------------------
``concretization_cache:size_limit``
-----------------------------------

Sets a limit on the size of the concretization cache in bytes. The limit is evaluated
after each concretization run; if Spack has stored more results than the limit allows, the
oldest concretization results are pruned until 10% of the limit has been removed.

> Setting this value to 0 or -1 disables the automatic pruning. It is expected users will be

Reviewer: Same here, let's just use `0` and reflect that in the schema
